clang/test/Driver/sycl-offload-intelfpga.cpp

```diff
@@ -211,9 +211,9 @@
 // RUN:  %clang_cl -fsycl -fintelfpga -foffload-static-lib=%t_aoco.a -### %s 2>&1 \
 // RUN:  | FileCheck -check-prefixes=CHK-FPGA-AOCO,CHK-FPGA-AOCO-WIN %s
 // CHK-FPGA-AOCO-LIN: clang-offload-bundler{{.*}} "-type=ao" "-targets=sycl-fpga_aoco-intel-unknown-sycldevice" "-inputs=[[INPUTLIB:.+\.a]]" "-check-section"
-// CHK-FPGA-AOCO-LIN: ld{{.*}} "-r" "-o" "[[PARTLINKOBJ:.+\.o]]" "{{.*}}crt1.o" "{{.*}}crti.o" "[[INPUTLIB]]" "{{.*}}crtn.o"
+// CHK-FPGA-AOCO-LIN: ld{{.*}} "-r" "-o" "[[PARTLINKOBJ:.+\.o]]" "{{.*}}crt1.o" "{{.*}}crti.o" "{{.*}}.o" "[[INPUTLIB]]" "{{.*}}crtn.o"
 // CHK-FPGA-AOCO-LIN: clang-offload-bundler{{.*}} "-type=oo" "-targets=sycl-spir64_fpga-unknown-unknown-sycldevice" "-inputs=[[PARTLINKOBJ]]" "-outputs={{.*}}" "-unbundle"
-// CHK-FPGA-AOCO-WIN: clang-offload-bundler{{.*}} "-type=aoo" "-targets=sycl-spir64_fpga-unknown-unknown-sycldevice-coff" "-inputs=[[INPUTLIB:.+\.a]]" "-outputs={{.*}}" "-unbundle"                               
+// CHK-FPGA-AOCO-WIN: clang-offload-bundler{{.*}} "-type=aoo" "-targets=sycl-spir64_fpga-unknown-unknown-sycldevice-coff" "-inputs=[[INPUTLIB:.+\.a]]" "-outputs={{.*}}" "-unbundle"
```

Comment:
Am I right to assume this command is affected by the fix? If so, checking the object filename's presence here would be preferable.

Reply:
This command is not affected here.  The problem with source to partial-link is a Linux specific change.  Windows static offloading does separate unbundlings of archive and object.

Reply:
Ah, I see, thanks for clearing this out. I initially thought that the Linux line required an update, but this can be disregarded now.

Reply:
BTW, what is the cause for a different MSVC flow? (Sorry if this has already been answered earlier & I'm just forgetting stuff)

Reply:
No change to the flow.  That line change just cleans up trailing whitespace (found from clang-format)

Reply:
Yep, I understand this :) I'm simply inquiring as to why the arhive & object unbundling are separated for Windows, while the partial link step is used for Linux - no relation to the patch itself.

Reply:
Oh, I misunderstood your question.  The reason the path is different is due to the fact that Windows doesn't support the partial linking concept, so we have to unbundle the entire archive independently similar to what we do with fat objects.

Reply:
Got it, thanks.